Custom CA certificates (#913)
epiphany-platform/epiphany: docs/home/howto/PREREQUISITES.md

*Note: Epicli will only run on Lixux or MacOS and not on Windows. This is because Ansible at this point in time does not work on Windows.*

*Note: You might want to consider installing Epicli in a virtual python enviroment. More information can be found [here.](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/)*

1. To be able to run the Epicli from your local OS you have to install:

    - Python 3.7
    - PIP

2. Install the following Python dependencies (in your virtual environment) using PIP:

    ```bash
    pip install wheel setuptools twine
    ```

3. Open a terminal in `/core/src/epicli` and run the following to build the Epicli wheel:

    ```bash
    ./build-wheel.sh
    ```

4. Install the Epicli wheel:

    ```bash
    pip install dist/epicli-VERSION-py3-none-any.whl
    ```

5. Verify the Epicli installation:

    ```bash
    epicli --version
    ```

    This should return the version of the CLI deployed.

Now you can use Epicli directly on your machine.

## Epicli development

For setting up en Epicli development environment please refer to this dedicated document [here.](./../DEVELOPMENT.md)

> ## Imporant notes

Reviewer: ```suggestion
## Important notes
```

  ```bash
  docker run -it -v POSSIBLE_MOUNTS... -e HTTP_PROXY=http://PROXY_SERVER:PORT -e HTTPS_PROXY=http://PROXY_SERVER:PORT http_proxy=http://PROXY_SERVER:PORT -e https_proxy=http://PROXY_SERVER:PORT --rm IMAGE_NAME
  ```

### Note about custom CA certificates

In some cases it might be that a company uses custom CA certificates for providing secure connections. To use these with Epicli you can do the following:

#### Devcontainer

> Before building the VSCode devcontainer place the *.crt file here: `/epiphany/core/src/epicli/.devcontainer/cert/`. Then the certificat will be include and configured during the build process. After that no additional configuration should be needed.

Reviewer: ```suggestion
Before building the VSCode devcontainer place the *.crt file here: `/epiphany/core/src/epicli/.devcontainer/cert/`. Then the certificate will be included and configured during the build process. After that no additional configuration should be needed.
```

#### MacOS

> Install the certiciate in you keychain as described [here.](https://www.sslsupportdesk.com/how-to-import-a-certificate-into-mac-os/)

Reviewer: ```suggestion
Install the certiciate in your keychain as described [here](https://www.sslsupportdesk.com/how-to-import-a-certificate-into-mac-os/).
```

#### Epicli container or Debian based OS

> If you are running Epicli from one of the prebuild containers or a debian based OS directly you can do the following to install the certificate:

Reviewer: ```suggestion
If you are running Epicli from one of the prebuild containers or a Debian based OS directly you can do the following to install the certificate:
```

  ```bash
  cp ./path/to/cert.crt /usr/local/share/ca-certificates/
  chmod 644 /usr/local/share/ca-certificates/cert.crt
  update-ca-certificates

> 

Reviewer: Closing backticks are missing for bash block

> *Note: Configuring the CA cert on the prebuild container only works on the debian based ones on NOT on alpine based ones.*

Reviewer: ```suggestion
*Note: Configuring the CA cert on the prebuild container only works on the `debian` based ones and NOT on `alpine` based.*
```
